src/main/java/frc/robot/RobotContainer.java

```diff
@@ -134,5 +138,26 @@ else if (selectorOne == 1 || selectorTwo == 1) {
         // Also return null if this ever gets to here because safety
         return null;
     }
+
+    public void joystickRumble(double leftValue, double rightValue) {
+        //Joystick rumble (driver feedback). leftValue/rightValue sets vibration force.
+        driverStick.setRumble(RumbleType.kLeftRumble, leftValue);
+        driverStick.setRumble(RumbleType.kRightRumble, rightValue);
+    }
+
+    /**
+     * Initialize the current RobotContainer instance
+     */
+    public static void init() {
+        if (currentInstance == null) {
+            currentInstance = new RobotContainer();
+        }
+    }
+
+    public static RobotContainer getInstance() {
```

Comment:
If you decide to change RobotContainer implementation to follow the singleton pattern then you should make the basic constructor (public RobotContainer() ) private, ensure you change the current creation of RobotContainer  to use the  getInstance (or we will have multiple RobotContainers).

To keep your changes simple, perhaps leave RobotContainer initialization as is and just add the joystick rumble parts

Reply:
True, that's a good point. No need to reinvent what has already been done.